firecrest/BasicClient.py

```diff
@@ -637,7 +711,66 @@ def extract(self, machine: str, source_path: str, target_path: str, extension: s
                 "extension": extension
             },
         )
-        self._json_response([resp], 201)
+        timeout_str = "Command has finished with timeout signal"
+        if (
+            resp.status_code == 201 or
+            fail_on_timeout or
+            resp.status_code != 400 or
+            resp.json().get('error', '') != timeout_str
+        ):
+            self._json_response([resp], 201)
+        else:
+            logger.debug(
+                f"Extraction of {source_path} to {target_path} has finished "
+                f"with timeout signal. Will submit a job to extract the "
+                f"file."
+            )
+
+            job_info = self.submit_extract_job(
+                machine,
+                source_path,
+                target_path,
+                extension
+            )
+            jobid = job_info['jobid']
+            active_jobs = self.poll_active(
+                machine,
+                [jobid]
+            )
+            intervals = itertools.cycle(
+                [1, 1, 1, 1, 1,
+                 5, 5, 5, 5,
+                 10, 10, 10,
+                 30]
+            )
+            while (
+                active_jobs and
+                not slurm_state_completed(active_jobs[0]['state'])
+            ):
+                time.sleep(next(intervals))
+                active_jobs = self.poll_active(
+                    machine,
+                    [jobid]
+                )
+
+            if (
+                active_jobs and
+                active_jobs[0]['state'] != 'COMPLETED'
+            ):
```

Comment:
This condition will always be `False`

```suggestion
            if (
                active_jobs[0]['state'] != 'COMPLETED'
            ):
```